packages/theme/modules/catalog/pages/product.vue

```diff
@@ -560,9 +572,48 @@ export default defineComponent({
 
       fetchProducts();
     };
+
+    const addToCart = async () => {
+      await addItem({ product: product.value, quantity: qty.value });
+
+      if (
+        cameFromWishlistPage.value
+        && isProductInWishlist.value
+        && !cartError.value?.addItem
+      ) {
+        await removeItemFromWishlist({ product: product.value });
+
+        sendNotification({
+          id: Symbol('product_added_to_cart_from_wishlist_product_page'),
+          message: i18n.t(`You added ${getProductName(
+            product.value,
+          )} to your shopping cart.`) as string,
+          type: 'success',
+          icon: 'check',
+          persist: false,
+          title: 'Wishlist',
+        });
+      } else if (cartError.value.addItem.message) {
+        sendNotification({
+          id: Symbol('product_added_to_cart_from_wishlist_error_product_page'),
+          message: cartError.value.addItem.message,
+          type: 'danger',
+          icon: 'cross',
+          persist: false,
+          title: 'Wishlist error',
+        });
+      }
+    };
+
+    onMounted(() => {
+      if (from.value?.path === localePath('/customer/my-wishlist')) {
```

Comment:
1. Wondering if it wouldn't be easier to just redirect to the product page with some query parameter `/p/product?from-wishlist` - that way if someone refreshes the page the "fromWishlist" information won't be lost

2. Shouldn't this be a computed?

3. If the query paramater doesn't work, it'll be easier to maybe use the route *name*, not the route path

```suggestion
      if (from.value?.name === localeRoute({ name: 'customer-my-wishlist' })) {
```

(see modules/customer/index.ts for route name)